test/e2e/connectivity_test.go

```diff
@@ -136,34 +147,73 @@ func TestHostPortPodConnectivity(t *testing.T) {
 	data.testHostPortPodConnectivity(t)
 }
 
-// createPodsOnDifferentNodes creates numPods busybox test Pods and assign them to all the different
-// Nodes in round-robin fashion, then returns the names of the created Pods as well as a function
-// which will delete the Pods when called.
-func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, cleanup func()) {
-	podNames = make([]string, 0, numPods)
+// createPodsOnDifferentNodes creates numPods agnhost test Pods and assign them to Nodes in the following fashion. This
+// function returns the names of the created Pods as well as a function which will delete the Pods when called.
+func createPodsOnDifferentNodes(t *testing.T, data *TestData, numPods int) (podNames []string, isWindows []bool, cleanup func() error) {
```

Comment:
my suggestion was also to get rid of the `numPods` parameter and just return a list of Pods with a length equal to the number of Nodes in the cluster. The list of Pods can be ordered as follows: linux Pod, windows Pod, linux Pod, windows Pod, ... I think this will be nicer than having a second `isWindows` slice? We can also remove `choosePodsOnDifferentNodes`. The caller of `createPodsOnDifferentNodes` can then choose to use the correct number of Pods for its usage and is guaranteed to have a mix of Linux & Windows Pods when applicable.

Reply:
Please correct me if I misunderstand something.
As for `isWindows`, if the testbed has 2 windows node and 1 linux node, then the Pod list will be like: [lin, win, win]? Then without `isWindows` slice, we cannot choose correct commands and options for these Pod on different OS.

Reply:
Ok, maybe we can just leave an empty string there: [lin, win, "", win]?

Reply:
Yes, I think podInfo works. Thank you!

Reply:
Updated.